lambdas/evaluate-gpg45-scores/src/main/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandler.java

```diff
@@ -129,13 +135,27 @@ public Map<String, Object> handleRequest(JourneyRequest event, Context context)
                 return JOURNEY_VCS_NOT_CORRELATED.toObjectMap();
             }
 
-            return checkForMatchingGpg45Profile(
+            boolean hasMatchingGpg45Profile =
+                    checkForMatchingGpg45Profile(
                             vcStoreItems,
                             ipvSessionItem,
                             clientOAuthSessionItem,
                             credentials,
-                            ipAddress)
-                    .toObjectMap();
+                            ipAddress);
+
+            if (configService.enabled(CoreFeatureFlag.INHERITED_IDENTITY)
+                    && hasMatchingGpg45Profile) {
+                for (VcStoreItem vcStoreItem : allVcStoreItems) {
+
+                    if (vcStoreItem.getCredentialIssuer().equals(HMRC_MIGRATION_CRI)) {
+                        verifiableCredentialService.deleteVcStoreItem(
+                                vcStoreItem.getUserId(), vcStoreItem.getCredentialIssuer());
+                    }
```

Comment:
Could use `VcHelper.filterVCBasedOnProfileType(allVcStoreItems, ProfileType.OPERATIONAL_HMRC)` and then run the deletion if the result is non empty?

Reply:
I think in this case using the HMRC CRI might actually be better - we don't know whether other operational profiles will be weaker/stronger

Reply:
Does checking the issuer is HMRC differentiate between the `OPERATIONAL_HMRC` profile type? If they started using other profile types it'd still be the same issuer?